src/frontend/src/crypto/ed25519.ts

```diff
@@ -0,0 +1,101 @@
+import { Ed25519KeyIdentity } from "@dfinity/identity";
+import { mnemonicToSeedSync } from "bip39";
+
+// A constant used for xor-ing derived paths to make them hardened.
+const HARDENED = 0x80000000;
+
+/**
+ * Create an Ed25519 according to SLIP 0010:
+ * https://github.com/satoshilabs/slips/blob/master/slip-0010.md
+ *
+ * The derivation path is an array that is always interpreted as a hardened path.
+ * e.g. to generate m/44'/223’/0’/0’/0' the derivation path should be [44, 223, 0, 0, 0]
+ */
+export async function fromSeedWithSlip0010(
+  masterSeed: Uint8Array,
+  derivationPath?: number[]
+): Promise<Ed25519KeyIdentity> {
+  let [slipSeed, chainCode] = await generateMasterKey(masterSeed);
+
+  if (derivationPath) {
+    for (let i = 0; i < derivationPath.length; i++) {
+      [slipSeed, chainCode] = await derive(
+        slipSeed,
+        chainCode,
+        derivationPath[i] | HARDENED
+      );
+    }
+  }
+
+  return Ed25519KeyIdentity.generate(slipSeed);
+}
+
+/**
+ * Create an Ed25519 based on a mnemonic phrase according to SLIP 0010:
+ * https://github.com/satoshilabs/slips/blob/master/slip-0010.md
+ *
+ * The derivation path is an array that is always interpreted as a hardened path.
+ * e.g. to generate m/44'/223’/0’/0’/0' the derivation path should be [44, 223, 0, 0, 0]
+ */
+export async function fromMnemonic(
+  mnemonic: string,
+  derivationPath?: number[]
```

Comment:
Could we put our derivation path somewhere as a constant?

Reply:
This is meant to be a generic API and shouldn't be coupled to a specific use-case. If there's a flow that needs a specific path, then I'd argue that the definition of that path should be part of that flow, not here.

Reply:
Got it. We'll find a good spot in the codebase for it then.